app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -26,10 +26,24 @@
     var uniqueId = _.uniqueId();
 
     ctrl.setFocusClass = 'edit-environment-from-set-focus-' + uniqueId;
+    ctrl.alerts = {};
+
+    ctrl.isImageSecret = function(secret) {
+      if (_.has(secret, ".dockerconfigjson") || _.has(secret, ".dockercfg")) {
+        return true;
+      }
+      return false;
+    };
 
     ctrl.viewOverlayPanel = function(entry) {
       ctrl.decodedData = entry.data;
       ctrl.overlayPaneEntryDetails = entry;
+      if (ctrl.isImageSecret(ctrl.decodedData)) {
+        ctrl.alerts["cantInjectSecret"] = {
+          type: "warning",
+          message: "This secret keys won't be infected into the deployment pod."
```

Comment:
It's not just image secrets. It's secrets that have keys with names that are not valid env vars. See

https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/addConfigToApplication.js#L63-L66

We should warn about what keys specifically won't be added using the regex. I think maybe a warning icon by the name in this dialog with tooltip text saying it's not a valid environment variable name.